docs/HTTP_V2_SPEC.md

```diff
@@ -1,106 +1,236 @@
 # The sqld HTTP API v2 specification ("Hrana over HTTP")
 
-Version 2 of the HTTP API ("Hrana over HTTP") extends the HTTP API with features
-introduced in Hrana 2.
+Version 2 of the HTTP API ("Hrana over HTTP") exposes stateful streams from
+Hrana over HTTP. It provides functionality equivalent to Hrana and it is useful
+for environments with missing or incomplete support for WebSockets.
+
+This version deprecates version 1 of the HTTP API. Both clients and servers
+should move to version 2 as soon as possible.
 
 ## Overview
 
-This HTTP API uses data structures and semantics from the Hrana 2 protocol.
-Endpoints in the HTTP API correspond to requests in Hrana. Each request is
-executed as if a fresh Hrana stream was opened for the request.
+The HTTP API uses data structures and semantics from the Hrana 2 protocol.
 
-All request and response bodies are encoded in JSON, with content type
-`application/json`.
+Individual requests on the same stream are tied together by the use of a baton.
+The server returns a baton in every response to a request on the stream, and the
+client then needs to include the baton in the subsequent request. The client
+must serialize the requests: it must wait for a response to the previous request
+before sending next request.
 
-## Version check
+The server can also optionally specify a different URL that the client should
+use for the requests on the stream. This can be used to ensure that stream
+requests are "sticky" and reach the same server.
 
-```
+The server will close streams after a short period of inactivity, to make sure
+that abandoned streams don't accumulate on the server.
+
+## Check support for version 2
+
+```typescript
 GET /v2
 ```
 
-A server that supports HTTP API v2 must return a success response for requests
-on path `/v2`. This can be used as a crude and unreliable mechanism for clients
-to fall back to version 1 of the protocol if the server does not support version
-2.
+If the server supports this version of the HTTP API, it should return a 2xx
+response for a GET request on `/v2`. This can be used as a crude version
+negotiation mechanism by the client.
 
-## Execute a statement
+## Execute requests on a stream
 
-```
-POST /v2/execute
+```typescript
+POST /v2/stream
```

Comment:
The reason for this name is that this sends requests to a Hrana stream. "execute" means "execute a single statement" in Hrana, so that might be confusing, too :/

Reply:
With this proposed API, no server-sent events are needed, these are standard HTTP requests, for maximum simplicity and compatibility.

Reply:
I renamed this endpoint to "pipeline", because it receives a pipeline of requests on a stream. Does that sound better to you, Athos? @athoscouto